big_tests/tests/gdpr_SUITE.erl

```diff
@@ -225,15 +278,190 @@ retrieve_roster(Config) ->
+
+            M1 = muc_light_helper:when_muc_light_message_is_sent(Alice, Room, BodyMucAlice, <<"Id1">>),
+            muc_light_helper:then_muc_light_message_is_received_by([Alice, Bob, Kate], M1),
+            M2 = muc_light_helper:when_muc_light_message_is_sent(Bob, Room, BodyMucBob, <<"Id2">>),
+            muc_light_helper:then_muc_light_message_is_received_by([Alice, Bob, Kate], M2),
+
+            mam_helper:wait_for_room_archive_size(Domain, Room, 3),
+
+            escalus:send(Alice, escalus_stanza:chat_to(Bob, BodyPmAlice)),
+            escalus:send(Bob, escalus_stanza:chat_to(Alice, BodyPmBob)),
+
+            [mam_helper:wait_for_archive_size(User, 5) || User <- [Alice, Bob]],
+            mam_helper:wait_for_archive_size(Kate, 3),
+
+            MamBackends = choose_mam_backend(Config, mam),
+            MamMucBackends = choose_mam_backend(Config, mam_muc),
+            maybe_stop_and_unload_module(mod_mam, MamBackends, Config),
+            maybe_stop_and_unload_module(mod_mam_muc, MamMucBackends, Config),
+
+            true = mongoose_helper:successful_rpc(mod_mam_meta, get_mam_module_opt,
+                                                   [domain(), mod_mam, archive_groupchats, undefined]),
+
+            AliceDir = retrieve_all_personal_data(Alice, Config),
+            BobDir = retrieve_all_personal_data(Bob, Config),
+            KateDir = retrieve_all_personal_data(Kate, Config),
+
+            validate_personal_data(
+                AliceDir, "mam_muc", ["id", "message"],
+                [#{"message" => [{contains, binary_to_list(BodyMucAlice)}]}], []),
+            validate_personal_data(
+                BobDir, "mam_muc", ["id", "message"],
+                [#{"message" => [{contains, binary_to_list(BodyMucBob)}]}], []),
+
+            AliceRoomJid = <<RoomJid/bitstring, "/", (escalus_client:short_jid(Alice))/bitstring>>,
+            BobRoomJid = <<RoomJid/bitstring, "/", (escalus_client:short_jid(Bob))/bitstring>>,
+
+            MucPM = [#{"message" => [{contains, "urn:xmpp:muclight:0#affiliations"}],
+                       "from" => [{jid, RoomJid}]},
+                     #{"message" => [{contains, BodyMucAlice}], "from" => [{jid, AliceRoomJid}]},
+                     #{"message" => [{contains, BodyMucBob}], "from" => [{jid, BobRoomJid}]}],
+            AllPM = MucPM ++ [#{"message" => [{contains, BodyPmAlice}],
+                                "from" => [{jid, escalus_client:full_jid(Alice)}]},
+                              #{"message" => [{contains, BodyPmBob}],
+                                    "from" => [{jid, escalus_client:full_jid(Bob)}]}],
+            SortFn = muc_msg_first(RoomJid),
+            validate_personal_data(AliceDir, "mam_pm", ["id", "from", "message"], AllPM, SortFn),
+            validate_personal_data(BobDir, "mam_pm", ["id", "from", "message"], AllPM, SortFn),
+            validate_personal_data(KateDir, "mam_pm", ["id", "from", "message"], MucPM, SortFn),
```

Comment:
Why not to use JSON and maps everywhere, because it's easier to work with.

And after just add json to csv converter, that can be covered with simple tests?